scalajslib/src/mill/scalajslib/ScalaJSModule.scala

```diff
@@ -73,22 +75,31 @@ trait ScalaJSModule extends scalalib.ScalaModule { outer =>
 
   def scalaJSToolsClasspath = T { scalaJSWorkerClasspath() ++ scalaJSLinkerClasspath() }
 
-  def fastOpt = T {
-    linkTask(FastOpt)()
+  def fastLinkJS: Target[Report] = T {
+    linkTask(optimize = false, legacy = false)()
   }
 
-  def fullOpt = T {
-    linkTask(FullOpt)()
+  def fullLinkJS: Target[Report] = T {
+    linkTask(optimize = true, legacy = false)()
   }
 
-  private def linkTask(mode: OptimizeMode): Task[PathRef] = T.task {
-    link(
-      worker = ScalaJSWorkerApi.scalaJSWorker(),
+  def fastOpt: Target[PathRef] = T {
+    linkTask(optimize = false, legacy = true)().publicModules.head.jsFile
+  }
+
+  def fullOpt: Target[PathRef] = T {
+    linkTask(optimize = true, legacy = true)().publicModules.head.jsFile
+  }
```

Comment:
Didn't deprecate these two methods yet because I want to see if the new methods cover all people use cases.
I will deprecate them in a future PR.
I made `fastLinkJS` and `fullLinkJS` work with Scala.js 1.2- so we can safely delete them in Mill 0.11 without dropping support for Scala.js 0.6